gcexport.py

```diff
@@ -590,17 +657,17 @@ def extract_device(device_dict, details, start_time_seconds, args, http_caller,
     metadata = details['metadataDTO']
     device_app_inst_id = metadata['deviceApplicationInstallationId'] if present('deviceApplicationInstallationId', metadata) else None
     if device_app_inst_id:
-        if not device_dict.has_key(device_app_inst_id):
+        if device_app_inst_id not in device_dict:
             # observed from my stock of activities:
             # details['metadataDTO']['deviceMetaDataDTO']['deviceId'] == null -> device unknown
             # details['metadataDTO']['deviceMetaDataDTO']['deviceId'] == '0' -> device unknown
             # details['metadataDTO']['deviceMetaDataDTO']['deviceId'] == 'someid' -> device known
             device_dict[device_app_inst_id] = None
             device_meta = metadata['deviceMetaDataDTO'] if present('deviceMetaDataDTO', metadata) else None
             device_id = device_meta['deviceId'] if present('deviceId', device_meta) else None
-            if not device_meta.has_key('deviceId') or device_id and device_id != '0':
+            if 'deviceId' not in device_meta or device_id and device_id != '0':
                 device_json = http_caller(URL_GC_DEVICE + str(device_app_inst_id))
-                file_writer(args.directory + '/device_' + str(device_app_inst_id) + '.json',
+                file_writer(args.directory + sep + 'device_' + str(device_app_inst_id) + '.json',
```

Comment:
```suggestion
                file_writer(join(args.directory, f'device_{device_app_inst_id}.json'),
```

Reply:
Note: I used join() as this is how you imported the function. However, it should be `os.path.join` as its more readable.

Reply:
If I understand correctly (having found only https://stackoverflow.com/a/46182112/3686) it's a bit cumbersome to use f-strings with Python 2 (must install `future-fstrings` with pip etc), meaning complicating the setup instructions for systems with Python 2. Or am I missing something?

Reply:
You are right. I'm sorry, forgot the f-strings are not supported in Python2 by default.
There are few more readable options for string concatenations, however by using `+` operator is usually much quicker. As the project is not large, lets keep it as you have already proposed in the pull request.

Reply:
You are right. I'm sorry, forgot the f-strings are not supported in Python2 by default.
There are few more readable options for string concatenations, however by using + operator is usually much quicker. As the project is not large, lets keep it with the `+` operator.
However there is still way for improvements
```suggestion
                file_writer(join(args.directory, 'device_' + str(device_app_inst_id) + '.json'),
```